cpp/src/strings/convert/convert_datetime.cu

```diff
@@ -254,21 +255,29 @@ struct parse_datetime
                 case 'p':
                 {
                     string_view am_pm(ptr,2);
-                    if( (timeparts[TP_HOUR] <= 12) &&
-                        ((am_pm.compare("PM",2)==0) || (am_pm.compare("pm",2)==0)) )
-                        timeparts[TP_HOUR] += 12;
+                    auto hour = timeparts[TP_HOUR];
+                    if((am_pm.compare("AM",2)==0) || (am_pm.compare("am",2)==0))
+                    {
+                        if( hour == 12 )
+                            hour = 0;
+                    }
+                    else if( hour < 12 )
+                        hour += 12;
+                    timeparts[TP_HOUR] = hour;
                     break;
                 }
                 case 'z':
                 {
-                    int sign = *ptr=='-' ? -1:1;
+                    int sign = *ptr=='-' ? 1:-1;
```

Comment:
:+1: confused me at first, but makes sense. maybe add a comment as to why this _seems_ counter-intuitive?